packages/algob/src/lib/tx.ts

```diff
@@ -36,11 +36,25 @@ export async function mkSuggestedParams (
 export function makeAssetCreateTxn (
   name: string, asaDef: ASADef, flags: ASADeploymentFlags, txSuggestedParams: tx.SuggestedParams
 ): tx.Transaction {
+  // Load Note
+  // Load payFlags.note (ignored if payFlags.noteb64 is present)
+  // undefined if none of them is present.
+  // If TxParams has noteb64 or note , it gets precedence
   const encoder = new TextEncoder();
+  let note;
+  if (flags.noteb64 ?? flags.note) {
+    // TxParams note
+    note = flags.noteb64 ? encoder.encode(flags.noteb64) : encoder.encode(flags.note);
+  } else {
+    // ASA definition note
+    note = asaDef.noteb64 ? encoder.encode(asaDef.noteb64)
+      : (asaDef.note ? encoder.encode(asaDef.note) : undefined);
```

Comment:
let's create and use a function (because we have a repetition )
```
note = decodeNote(note, note64)
```

Reply:
Also, please add a test to that function.

Reply:
did you mean encodeNote ?

Reply:
Done created encodeNote function with test